lib/mastodon/cli/email_domain_blocks.rb

```diff
@@ -5,9 +5,33 @@
 
 module Mastodon::CLI
   class EmailDomainBlocks < Base
+    option :only_blocked, type: :boolean, defaut: false
+    option :only_with_approval, type: :boolean, default: false
     desc 'list', 'List blocked e-mail domains'
+    long_desc <<-LONG_DESC
+      By default this command lists all domains in the email domain block list
+      and their associated MX records (if included).
+
+      If the --only-blocked option is provided, this command will list only email
+      domains that are fully blocked from signup.
+
+      If the --only-with-approval option is provided, this command will list only
+      email domains that are allowed to be used but require manual approval.
+
+      The --only-blocked and --only-with-approval options are mutually exclusive.
+    LONG_DESC
     def list
-      EmailDomainBlock.parents.find_each do |parent|
+      fail_with_message 'Cannot specify both --only-blocked and --only-with-approval' if options[:only_blocked] && options[:only_with_approval]
+
+      base_query = EmailDomainBlock.where(parent_id: nil)
```

Comment:
The `parents` which was removed adds those conditions?

Reply:
Didn't notice that I had deleted it. Restored.